core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/GlobalOrdinalsSignificantTermsAggregator.java

```diff
@@ -131,6 +131,9 @@ public SignificantStringTerms buildAggregation(long owningBucketOrdinal) throws
             // global stats
             spare.updateScore(significanceHeuristic);
             spare = ordered.insertWithOverflow(spare);
+            if (spare == null) {
+                consumeBucketsAndMaybeBreak(1);
+            }
```

Comment:
Should there be an else here to remove the buckets inside spare if its not null?

Reply:
There is no bucket inside spare yet. We are just selecting the top terms here. When the selection is done we build the bucket aggregation for each top terms only so we'll count the inner buckets only for the final top terms.